shell/browser/extensions/api/runtime/electron_runtime_api_delegate.cc

```diff
@@ -42,10 +43,57 @@ bool ElectronRuntimeAPIDelegate::CheckForUpdates(
 void ElectronRuntimeAPIDelegate::OpenURL(const GURL& uninstall_url) {}
 
 bool ElectronRuntimeAPIDelegate::GetPlatformInfo(PlatformInfo* info) {
-  // TODO(nornagon): put useful information here.
-#if defined(OS_LINUX)
-  info->os = api::runtime::PLATFORM_OS_LINUX;
-#endif
+  const char* os = update_client::UpdateQueryParams::GetOS();
+  if (strcmp(os, "mac") == 0) {
+    info->os = extensions::api::runtime::PLATFORM_OS_MAC;
+  } else if (strcmp(os, "win") == 0) {
+    info->os = extensions::api::runtime::PLATFORM_OS_WIN;
+  } else if (strcmp(os, "linux") == 0) {
+    info->os = extensions::api::runtime::PLATFORM_OS_LINUX;
+  } else if (strcmp(os, "openbsd") == 0) {
+    info->os = extensions::api::runtime::PLATFORM_OS_OPENBSD;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  const char* arch = update_client::UpdateQueryParams::GetArch();
+  if (strcmp(arch, "arm") == 0) {
+    info->arch = extensions::api::runtime::PLATFORM_ARCH_ARM;
+  } else if (strcmp(arch, "arm64") == 0) {
+    info->arch = extensions::api::runtime::PLATFORM_ARCH_ARM64;
+  } else if (strcmp(arch, "x86") == 0) {
+    info->arch = extensions::api::runtime::PLATFORM_ARCH_X86_32;
+  } else if (strcmp(arch, "x64") == 0) {
+    info->arch = extensions::api::runtime::PLATFORM_ARCH_X86_64;
+  } else if (strcmp(arch, "mipsel") == 0) {
+    info->arch = extensions::api::runtime::PLATFORM_ARCH_MIPS;
+  } else if (strcmp(arch, "mips64el") == 0) {
+    info->arch = extensions::api::runtime::PLATFORM_ARCH_MIPS64;
+  } else {
+    NOTREACHED();
+    return false;
+  }
+
+  const char* nacl_arch = update_client::UpdateQueryParams::GetNaclArch();
+  if (strcmp(nacl_arch, "arm") == 0) {
+    info->nacl_arch = extensions::api::runtime::PLATFORM_NACL_ARCH_ARM;
+  } else if (strcmp(nacl_arch, "arm64") == 0) {
+    // Use ARM for ARM64 NaCl, as ARM64 NaCl is not available.
+    info->nacl_arch = extensions::api::runtime::PLATFORM_NACL_ARCH_ARM;
+  } else if (strcmp(nacl_arch, "x86-32") == 0) {
+    info->nacl_arch = extensions::api::runtime::PLATFORM_NACL_ARCH_X86_32;
+  } else if (strcmp(nacl_arch, "x86-64") == 0) {
+    info->nacl_arch = extensions::api::runtime::PLATFORM_NACL_ARCH_X86_64;
+  } else if (strcmp(nacl_arch, "mips32") == 0) {
+    info->nacl_arch = extensions::api::runtime::PLATFORM_NACL_ARCH_MIPS;
+  } else if (strcmp(nacl_arch, "mips64") == 0) {
+    info->nacl_arch = extensions::api::runtime::PLATFORM_NACL_ARCH_MIPS64;
+  } else {
+    NOTREACHED();
+    return false;
+  }
```

Comment:
We never support NaCl. Can we change this to:
```suggestion
  info->nacl_arch = extensions::api::runtime::PLATFORM_NACL_ARCH_NONE;
```

Reply:
So the extensions code doesn't seem to like using this value. By doing so, the renderer crashes and devtools gets disconnected. 😞

I reverted the suggested changes, but removed unsupported platforms (mips, mips64).

Reply:
hm, ok. /shrug then